contracts-proc/src/interface_id.rs

```diff
@@ -0,0 +1,123 @@
+//! Defines the `#[interface_id]` procedural macro.
+
+use std::mem;
+
+use convert_case::{Case, Casing};
+use proc_macro::TokenStream;
+use proc_macro2::Ident;
+use quote::quote;
+use syn::{
+    parse::{Parse, ParseStream},
+    parse_macro_input, FnArg, ItemTrait, LitStr, Result, Token, TraitItem,
+};
+
+/// Computes an interface id as an associated constant for the trait.
+pub(crate) fn interface_id(
+    _attr: &TokenStream,
+    input: TokenStream,
+) -> TokenStream {
+    let mut input = parse_macro_input!(input as ItemTrait);
+    let mut output = quote! {};
+
+    let mut selectors = Vec::new();
+    for item in &mut input.items {
+        let TraitItem::Fn(func) = item else {
+            continue;
+        };
+
+        let mut override_name = None;
+        for attr in mem::take(&mut func.attrs) {
+            let Some(ident) = attr.path().get_ident() else {
+                func.attrs.push(attr);
+                continue;
+            };
+            if *ident == "selector" {
```

Comment:
Here we retrieve all attributes of type: `#[selector(name = "actualSelectorName")` in case actual selector's name should be different for abi